source/containers/document-pii-detection/parsers/html_parser.py

```diff
@@ -0,0 +1,152 @@
+
+import re
+import six
+
+from bs4 import BeautifulSoup
+
+from .parser import BaseParser
+
+class HtmlParser(BaseParser):
+    def __init__(self, s3_client):
+        super().__init__(s3_client=s3_client)
+        # additional PdfParser constructor code here
+
+    def parse_file(self, html_path):
+        """
+        Extracts text from a html file and returns a string of content.
+        """
+
+        with open(html_path, "rb") as stream:
+            soup = BeautifulSoup(stream, 'lxml')
+
+        # Convert tables to ASCII ones
+        soup = self._replace_tables(soup)
+
+        # Join inline elements
+        soup = self._join_inlines(soup)
+
+        # Make HTML
+        html = ''
+        elements = soup.find_all(True)
+        elements = [el for el in filter(self._visible, elements)]
+        for elem in elements:
+            string = elem.string
+            if string is None:
+                string = self._find_any_text(elem)
+            string = string.strip()
+            if len(string) > 0:
+                html += "\n" + string + "\n"
+        return [html]
+
+    _disallowed_names = [
+        'style', 'script', '[document]', 'head', 'title', 'html', 'meta',
+        'link', 'body',
+    ]
+
+    _inline_tags = [
+        'b', 'big', 'i', 'small', 'tt', 'abbr', 'acronym', 'cite', 'code',
+        'dfn', 'em', 'kbd', 'strong', 'samp', 'var', 'a', 'bdo', 'br', 'img',
+        'map', 'object', 'q', 'script', 'span', 'sub', 'sup', 'button',
+        'input', 'label', 'select', 'textarea',
+    ]
+
+    def _visible(self, element):
+        """Used to filter text elements that have invisible text on the page.
+        """
+        if element.name in self._disallowed_names:
+            return False
+        elif re.match(u'<!--.*-->', six.text_type(element.extract())):
+            return False
+        return True
+
+    def _inline(self, element):
+        """Used to check whether given element can be treated as inline
+        element (without new line after).
+        """
+        if element.name in self._inline_tags:
+            return True
+        return False
+
+    def _find_any_text(self, tag):
+        """Looks for any possible text within given tag.
+        """
+        text = ''
+        if tag is not None:
+            text = six.text_type(tag)
+            text = re.sub(r'(<[^>]+>)', '', text)
+            text = re.sub(r'\s', ' ', text)
+            text = text.strip()
+        return text
+
+    def _parse_tables(self, soup):
+        """Returns array containing basic informations about tables for ASCII
+        replacement (look: _replace_tables()).
+        """
+        tables = []
+        for t in soup.find_all('table'):
+            t_dict = {'width': 0, 'table': t, 'trs': [], 'col_width': {}}
+            trs = t.find_all('tr')
+            if len(trs) > 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)

Similar issue at line number 93.

Reply:
Resolve by using if trs.